libs/entitlement/qa/entitlement-test-plugin/src/main/java/org/elasticsearch/entitlement/qa/test/URLConnectionFileActions.java

```diff
@@ -30,6 +31,17 @@ private static void withJdkFileConnection(CheckedConsumer<URLConnection, Excepti
         }
     }
 
+    private static void withJarConnection(CheckedConsumer<JarURLConnection, Exception> connectionConsumer) throws Exception {
+        var conn = EntitledActions.createJarURLConnection();
+        // Be sure we got the connection implementation we want
+        assert JarURLConnection.class.isAssignableFrom(conn.getClass());
+        try {
+            connectionConsumer.accept((JarURLConnection) conn);
+        } catch (IOException e) {
+            // It's OK, it means we passed entitlement checks, and we tried to perform some operation
```

Comment:
Can we actually use the APIs correctly, instead of expecting an IOException? If there is an exception we can't really be sure something else didn't go wrong (eg in entitlements that caused an IOException). Especially in the future if we add a NotEntitledIOException..

Reply:
I'll do it; in the case of JarURL should be simple enough.